File: src/main/java/investLee/platform/ecommerce/domain/entity/ProductCategoryEntity.java
package investLee.platform.ecommerce.domain.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "product_categories")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductCategoryEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    private ProductEntity product;

    @ManyToOne(fetch = FetchType.LAZY)
    private CategoryEntity category;

    private boolean isPrimary;
}

File: src/product/infrastructure/entities/Product_Category.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn } from "typeorm";

import { CategoryEntity } from "@category/infrastructure/entities";
import ProductEntity from "./Product.entity";

@Entity("product_categories")
export default class ProductCategoryEntity {
  @PrimaryGeneratedColumn("increment", { type: "bigint" })
  id: number;

  @ManyToOne(() => ProductEntity, { onDelete: "CASCADE" })
  @JoinColumn({ name: "product_id" })
  product: ProductEntity;

  @ManyToOne(() => CategoryEntity, { onDelete: "CASCADE" })
  @JoinColumn({ name: "category_id" })
  category: CategoryEntity;

  @Column({ type: "boolean", default: false })
  is_primary: boolean;
}
